llpc/context/llpcCompiler.cpp

```diff
@@ -2091,144 +2017,85 @@ void Compiler::ReleaseContext(
 // It will try App's pipelince cache first if that's available.
 // Then try on the internal shader cache next if it misses.
 //
-// NOTE: Only two items in the array of shader caches; one for App's pipeline cache and one for internal cache
-#if LLPC_CLIENT_INTERFACE_MAJOR_VERSION < 38
+// Upon hit, Ready is returned and pElfBin is filled in. Upon miss, Compiling is returned and ppShaderCache and
+// phEntry are filled in.
 ShaderEntryState Compiler::LookUpShaderCaches(
-    IShaderCache*                    pAppPipelineCache, // [in]    App's pipeline cache
-    MetroHash::Hash*                 pCacheHash,        // [in]    Hash code of the shader
-    BinaryData*                      pElfBin,           // [inout] Pointer to shader data
-    ShaderCache**                    ppShaderCache,     // [in]    Array of shader caches.
-    CacheEntryHandle*                phEntry            // [in]    Array of handles of the shader caches entry
+    IShaderCache*                    pAppPipelineCache, // [in] App's pipeline cache
+    MetroHash::Hash*                 pCacheHash,        // [in] Hash code of the shader
+    BinaryData*                      pElfBin,           // [out] Pointer to shader data
+    ShaderCache**                    ppShaderCache,     // [out] Shader cache to use
+    CacheEntryHandle*                phEntry            // [out] Handle to use
     )
 {
-    ShaderEntryState cacheEntryState  = ShaderEntryState::New;
-    uint32_t         shaderCacheCount = 1;
-    Result           result           = Result::Success;
+    ShaderCache* pShaderCache[2];
+    uint32_t shaderCacheCount;
 
-    if (pAppPipelineCache != nullptr)
+    if (pAppPipelineCache != nullptr && cl::ShaderCacheMode != ShaderCacheForceInternalCacheOnDisk)
     {
-        ppShaderCache[0] = static_cast<ShaderCache*>(pAppPipelineCache);
-        ppShaderCache[1] = m_shaderCache.get();
+        // Put the application's cache last so that we prefer adding entries there (only relevant with old
+        // client version).
+        pShaderCache[0] = m_shaderCache.get();
+        pShaderCache[1] = static_cast<ShaderCache*>(pAppPipelineCache);
         shaderCacheCount = 2;
     }
     else
     {
-        ppShaderCache[0] = m_shaderCache.get();
-        ppShaderCache[1] = nullptr;
-    }
-
-    if (cl::ShaderCacheMode == ShaderCacheForceInternalCacheOnDisk)
-    {
-        ppShaderCache[0] = m_shaderCache.get();
-        ppShaderCache[1] = nullptr;
+        pShaderCache[0] = m_shaderCache.get();
         shaderCacheCount = 1;
     }
 
     for (uint32_t i = 0; i < shaderCacheCount; i++)
     {
-        cacheEntryState = ppShaderCache[i]->FindShader(*pCacheHash, true, &phEntry[i]);
+        // Lookup the shader. Allocate on miss when we've reached the last cache.
+        bool allocateOnMiss = (i + 1) == shaderCacheCount;
+        CacheEntryHandle hEntryCurrent;
```

Comment:
hEntryCurrent -> hCurrentEntry = nullptr
It is more natural to call it "current entry"